tests/integrations/raftstore/test_snap.rs

```diff
@@ -514,3 +523,143 @@ fn test_inspected_snapshot() {
+    let snap_dir = cluster.get_snap_dir(2);
+    let snap_mgr = cluster.get_snap_mgr(2);
+    let (mut snap_term, mut snap_index) = (0, 0);
+    let mut meta_path = String::new();
+    for entry in fs::read_dir(&snap_dir).unwrap() {
+        let entry = entry.unwrap();
+        let copy_to = entry.path().to_str().unwrap().replace("rev_1", "gen_2");
+        fs::copy(entry.path(), Path::new(&copy_to)).unwrap();
+        if copy_to.ends_with(".meta") {
+            let fname = copy_to.split("/").last().unwrap();
+            let prefix = fname.strip_suffix(".meta").unwrap();
+            let parts: Vec<_> = prefix.split('_').collect();
+            snap_term = parts[2].parse::<u64>().unwrap();
+            snap_index = parts[3].parse::<u64>().unwrap();
+            meta_path = copy_to;
+        }
+    }
+
+    // Keep sending snapshots to store 1.
+    let addr = cluster.sim.rl().get_addr(1);
+    let sec_mgr = cluster.sim.rl().security_mgr.clone();
+    let th = std::thread::spawn(move || {
+        loop {
+            for suffix in &[".meta", "_default.sst"] {
+                let mut src = PathBuf::from(&snap_dir);
+                src.push(&format!("gen_2_{}_{}{}", snap_term, snap_index, suffix));
+                let mut dst = PathBuf::from(&snap_dir);
+                dst.push(&format!("gen_2_{}_{}{}", snap_term, snap_index + 1, suffix));
+                fs::hard_link(&src, &dst).unwrap();
+            }
+
+            let snap_mgr = snap_mgr.clone();
+            let sec_mgr = sec_mgr.clone();
+            if let Err(e) = send_a_large_snapshot(
+                snap_mgr, sec_mgr, &addr, 2, snap_index, snap_term, &meta_path,
+            ) {
+                info!("send_a_large_snapshot fail: {}", e);
+                break;
+            }
+
+            meta_path = meta_path.replace(
+                &format!("{}.meta", snap_index),
+                &format!("{}.meta", snap_index + 1),
+            );
+            snap_index += 1;
+        }
+    });
+
+    // While store 1 keeps receiving snapshots, it should still can generate a snapshot on time.
+    pd_client.must_add_peer(rid, new_learner_peer(3, 3));
+    std::thread::sleep(Duration::from_millis(500));
+    must_get_equal(&cluster.get_engine(3), b"key-0000", b"value");
+
+    drop(cluster);
+    let _ = th.join();
+}
+
+fn send_a_large_snapshot(
+    mgr: SnapManager,
+    security_mgr: Arc<SecurityManager>,
+    addr: &str,
+    region_id: u64,
+    index: u64,
+    term: u64,
+    meta_path: &str,
+) -> std::result::Result<(), String> {
+    // Read a snapshot meta.
+    let mut meta_bytes = Vec::with_capacity(1024);
+    fs::File::open(meta_path)
+        .and_then(|mut f| f.read_to_end(&mut meta_bytes))
+        .unwrap();
+    let mut snap_meta = SnapshotMeta::default();
+    snap_meta.merge_from_bytes(&meta_bytes).unwrap();
+
+    // Construct a snapshot data.
+    let mut snap_data = RaftSnapshotData::default();
```

Comment:
Can the functions from snap.rs be reused?

Reply:
I guess no because `RaftSnapshotData` is constructed in `Snap::build`, which is not usefule here.